src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -75,8 +84,17 @@ export class KernelProcess implements IKernelProcess {
         this._kernelConnectionMetadata = kernelConnectionMetadata;
     }
     public async interrupt(): Promise<void> {
-        if (this.kernelDaemon) {
-            await this.kernelDaemon?.interrupt();
+        if (!this.canInterrupt) {
+            throw new Error('Kernel interrupt not supported in KernelProcess.ts');
+        }
+        if (this.pythonDaemon) {
+            traceInfo('Interrupting kernel via Daemon message');
+            await this.pythonDaemon.interrupt();
+        } else if (this._kernelConnectionMetadata.kernelSpec.interrupt_mode !== 'message' && this._process) {
+            traceInfo('Interrupting kernel via Signals');
+            this._process.kill('SIGINT');
```

Comment:
This only works on unix systems.
Oh wait, I need to ensure we check the Platform (if windows or not).
because windows doesn't supoort signals

Reply:
Doesn't this kill the process? Can you send a signal without 'kill'?

Reply:
On unix yes, on windows no.

Separate topic: I think .NET 6.0 has added support for signals on windows

Reply:
Oh I guess it just sends the signal. Kill with 'SIGTERM' is how you really kill. What a crappy method name.

Reply:
> oesn't this kill the process? Can you send a signal wi

That's how you send signals in node. weird isn't it.
Will update to make it more clear by using `process.kill(<pid>, signa)`